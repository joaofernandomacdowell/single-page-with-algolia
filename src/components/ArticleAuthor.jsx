import React from 'react';

const ArticleAuthor = ({ author }) =>
  <p className="zoom-article-author">{author.name}</p>

export default ArticleAuthor;