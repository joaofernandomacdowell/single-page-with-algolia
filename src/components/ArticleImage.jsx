import React from 'react';

const ArticleImage = ({ hit }) => {
  const { imageUrl, title, url } = hit;
  return (
    <a className="zoom-article-image__wrapper" href={url}>
      <figure className="zoom-article-image__figure">
        <img className="zoom-article-image__img"
          src={imageUrl}
          alt={title}
          title={title}
          loading="lazy"
        />
      </figure>
    </a>
  );
}

export default ArticleImage;