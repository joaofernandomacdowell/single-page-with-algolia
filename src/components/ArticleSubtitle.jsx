import React from 'react';

const ArticleSubtitle = ({ subtitle }) =>
  <p className="zoom-article-subtitle">{subtitle}</p>

export default ArticleSubtitle;