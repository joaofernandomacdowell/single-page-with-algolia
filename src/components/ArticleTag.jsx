import React from 'react';

const ArticleTag = ({ tags, url }) => {
  const mainTag = tags[0];
  const taghref = url.substring(0, url.lastIndexOf('/'));

  return (
    <a className="zoom-article-tag" href={taghref}>
      {mainTag}
    </a>
  );
}

export default ArticleTag;