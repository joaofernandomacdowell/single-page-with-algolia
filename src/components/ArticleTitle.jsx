import React from 'react';

const ArticleTitle = ({ title }) =>
  <h3 className="zoom-article-title">{title}</h3>

export default ArticleTitle;