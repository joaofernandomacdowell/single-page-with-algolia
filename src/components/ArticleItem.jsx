import React from 'react';

import ArticleImage from './ArticleImage';
import ArticleTag from './ArticleTag';
import ArticleTitle from './ArticleTitle';
import ArticleSubtitle from './ArticleSubtitle';
import ArticleAuthor from './ArticleAuthor';

const ArticleItem = ({ hit }) => (
  <div className="zoom-article-item__wrapper zoom-article-item__position" key={`hit-${hit.id}`}>
    <ArticleImage hit={hit} />
    <div className="zoom-article-item__description">
      <ArticleTag tags={hit.tags} url={hit.url} />
      <a href={hit.url}>
        <ArticleTitle title={hit.title} />
        <ArticleSubtitle subtitle={hit.subtitle} />
        <ArticleAuthor author={hit.author} />
      </a>
    </div>
  </div>
);

export default ArticleItem;