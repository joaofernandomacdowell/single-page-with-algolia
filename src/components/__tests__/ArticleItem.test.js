import React from 'react'
import { shallow } from 'enzyme';

import mockApiResponse from '../../helper';
import ArticleItem from '../ArticleItem';
import ArticleImage from '../ArticleImage';
import ArticleTag from '../ArticleTag';
import ArticleTitle from '../ArticleTitle';
import ArticleSubtitle from '../ArticleSubtitle';
import ArticleAuthor from '../ArticleAuthor';


describe('ArticleItem component', () => {
  const props = {
    hit: mockApiResponse.hits[0]
  };
  const wrapper = shallow(<ArticleItem {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with all required components', () => {
    // expect(wrapper.containsMatchingElement(ArticleImage)).toBe(true);
    // expect(wrapper.containsAnyMatchingElements([
    //   ArticleImage,
    //   ArticleTag,
    //   ArticleTitle,
    //   ArticleSubtitle,
    //   ArticleAuthor,
    // ])).toBe(true);
  });
});