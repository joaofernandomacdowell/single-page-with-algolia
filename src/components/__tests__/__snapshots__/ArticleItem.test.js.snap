// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem component match the snapshot 1`] = `ShallowWrapper {}`;
